lib/base/io-engine.hpp

```diff
@@ -174,40 +174,24 @@ class Timeout : public SharedObject
 	Timeout(boost::asio::io_context& io, Executor& executor, TimeoutFromNow timeoutFromNow, OnTimeout onTimeout)
 		: m_Timer(io)
 	{
-		Ptr keepAlive (this);
-
-		m_Cancelled.store(false);
 		m_Timer.expires_from_now(std::move(timeoutFromNow));
 
-		IoEngine::SpawnCoroutine(executor, [this, keepAlive, onTimeout](boost::asio::yield_context yc) {
-			if (m_Cancelled.load()) {
-				return;
-			}
-
-			{
-				boost::system::error_code ec;
-
-				m_Timer.async_wait(yc[ec]);
-
-				if (ec) {
-					return;
-				}
-			}
-
-			if (m_Cancelled.load()) {
-				return;
+		m_Timer.async_wait(boost::asio::bind_executor(executor, [onTimeout](boost::system::error_code ec) {
+			if (!ec) {
+				onTimeout();
 			}
+		}));
+	}
 
-			auto f (onTimeout);
-			f(std::move(yc));
-		});
+	~Timeout() override
+	{
+		Cancel();
```

Comment:
Please remove this you don't need it!
<img width="694" alt="Bildschirmfoto 2024-11-28 um 12 05 19" src="https://github.com/user-attachments/assets/97901656-c0ed-4032-aa47-a2894f098778">